pkg/kv/bulk/sst_batcher.go

```diff
@@ -365,27 +377,14 @@ func (b *SSTBatcher) doFlush(ctx context.Context, reason int) error {
 		b.stats.batchesDueToRange++
 	}
 
-	shouldSplit := false
-	if !b.disableSplits {
-		if b.lastRange.span.ContainsKey(start) && size >= b.lastRange.remaining {
-			// If this file is starting in the same span we last added to and is bigger
-			// than the size that range had when we last added to it, then we should
-			// split off the suffix of that range where this file starts and add it to
-			// that new range after scattering it.
-			log.VEventf(ctx, 2, "%s batcher splitting full range %s before adding file starting at %s",
-				b.name, b.lastRange.span, start)
-			shouldSplit = true
-		} else if reason == sizeFlush && !b.initialSplitDone && b.stats.batches == 1 {
```

Comment:
I _think_ we lost this split condition was this intentional?

Reply:
it's in a separate commit

Reply:
sorry didn't see the first commit message, just reviewed the whole blob